deepcell/image_generators/semantic.py

```diff
@@ -135,11 +141,13 @@ def __init__(self,
 
             for transform in transforms:
                 transform_kwargs = transforms_kwargs.get(transform, dict())
+                # add default dtype to kwargs
+
                 y_transform = _transform_masks(y_current, transform,
                                                data_format=data_format,
                                                **transform_kwargs)
                 if y_transform.shape[self.channel_axis] > 1:
-                    y_transform = np.asarray(y_transform, dtype='int32')
+                    y_transform = np.asarray(y_transform, dtype=self.y.dtype)
```

Comment:
Was there an issue where multi-channel y were getting forced to ints when they shouldn't?

Do we need this conditional anymore? Or should we just always use `self.y.dtype` regardless of the channel size of y?

Reply:
I think that issue was the bug in the HPS branch for doing online transforms, not 100% sure. I don't think I ran into anything similar.

We can't use self.y.dtype, because if the dtype is `int`, and we have inner-distance transform which produces floats from 0-1, it'll ruin the output. 

Maybe we could always use float? I vaguely remember one of these transforms expending ints for 0/1 equivalent of boolean, but I'm not sure